### ISSUE TITLE
BuildVariableTask fails when project name contains spaces (linux only)

### ISSUE THREAD

**Issue by schaettgenfr** [rfennell/AzurePipelines#1084]
When using the BuildVariableTask with a linux build in a project that contains a space, the task fails.
This is probably related to #947.

```
...
VERBOSE: Getting BuildDef for Build
VERBOSE: Using SystemVssConnection personal access token
##[error]MethodInvocationException: /home/vsts/work/_tasks/BuildVariableTask_115030de-9ced-4d2f-ad01-8c42d20d0327/2.3.19/BuildVariableTask.ps1:259
Line |
 259 |      $jsondata = $webclient.DownloadString($uri) | ConvertFrom-Json
     |      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Exception calling "DownloadString" with "1" argument(s): "The
     | remote server returned an error: (400) Bad Request."
...
```

There is no such error when using the task with a windows agent.
There seems to be a difference between powershell and pwsh with respect to arguments with quotes:

This is what it looks like running with a windows agent (additional newlines removed). Note the line "VERBOSE: teamproject":
```
Starting: BuildVariableTask
==============================================================================
Task         : Set variable on a build defintion
Description  : Set variable on a build defintion
Author       : Richard Fennell
Help         : Version: 2.3.19. [More Information](https://github.com/rfennell/vNextBuild/wiki/BuildTasks-Task/)
==============================================================================
Using executable 'powershell.exe'
powershell.exe D:\a\_tasks\BuildVariableTask_115030de-9ced-4d2f-ad01-8c42d20d0327\2.3.19\BuildVariableTask.ps1 -collectionUrl https://dev.azure.com/TrumpfCorp/ -token *** -teamproject 'TruTops Boost' -buildid 268613 -buildmode AllArtifacts -variable ServiceVersionProd -mode Manual -value 0.0.23 -usedefaultcreds False
(node:2672) Warning: Use Cipheriv for counter mode of aes-256-ctr
...
(node:2672) Warning: Use Cipheriv for counter mode of aes-256-ctr
VERBOSE: collectionUrl = [https://dev.azure.com/TrumpfCorp/]
VERBOSE: teamproject = [TruTops Boost]
VERBOSE: releaseid = []
VERBOSE: builddefid = []
VERBOSE: buildid = [268613]
VERBOSE: usedefaultcreds = False
VERBOSE: artifacts = []
VERBOSE: buildmode = [AllArtifacts]
VERBOSE: mode = [Manual]
VERBOSE: Running inside a build so updating current build 268613
VERBOSE: Checking API version available
VERBOSE: Getting BuildDef for Build
VERBOSE: Using SystemVssConnection personal access token
VERBOSE: API Version is 4.0
...
```

When running under linux, the output goes like this. Note the quotes in the line "VERBOSE: teamproject = ['TruTops Boost']":
```
Starting: BuildVariableTask
==============================================================================
Task         : Set variable on a build defintion
Description  : Set variable on a build defintion
Version      : 2.3.19
Author       : Richard Fennell
Help         : Version: 2.3.19. [More Information](https://github.com/rfennell/vNextBuild/wiki/BuildTasks-Task/)
==============================================================================
Using executable 'pwsh' as only only option on 'Linux'
pwsh /home/vsts/work/_tasks/BuildVariableTask_115030de-9ced-4d2f-ad01-8c42d20d0327/2.3.19\BuildVariableTask.ps1 -collectionUrl https://dev.azure.com/TrumpfCorp/ -token *** -teamproject 'TruTops Boost' -buildid 268611 -buildmode AllArtifacts -variable ServiceVersionProd -mode Manual -value 0.0.23 -usedefaultcreds False
(node:3215) Warning: Use Cipheriv for counter mode of aes-256-ctr
...
(node:3215) Warning: Use Cipheriv for counter mode of aes-256-ctr
VERBOSE: collectionUrl = [https://dev.azure.com/TrumpfCorp/]
VERBOSE: teamproject = ['TruTops Boost']
VERBOSE: releaseid = []
VERBOSE: builddefid = []
VERBOSE: buildid = [268611]
VERBOSE: usedefaultcreds = False
VERBOSE: artifacts = []
VERBOSE: buildmode = [AllArtifacts]
VERBOSE: mode = [Manual]
VERBOSE: Running inside a build so updating current build 268611
VERBOSE: Checking API version available
VERBOSE: Getting BuildDef for Build
VERBOSE: Using SystemVssConnection personal access token
VERBOSE: Getting BuildDef for Build
VERBOSE: Using SystemVssConnection personal access token
##[error]MethodInvocationException: /home/vsts/work/_tasks/BuildVariableTask_115030de-9ced-4d2f-ad01-8c42d20d0327/2.3.19/BuildVariableTask.ps1:259
Line |
 259 |      $jsondata = $webclient.DownloadString($uri) | ConvertFrom-Json
     |      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Exception calling "DownloadString" with "1" argument(s): "The
     | remote server returned an error: (400) Bad Request."
...
```

This very much looks like the quotes are leaking into the script on linux, and from there in a URL, causing the bad request response.

**Comment by rfennell** [comment]
Seems you have spotted the problem, I will  try to sort out a fix

**Comment by rfennell** [comment]
This has been fixed 2.4.5 which is going through the release pipeline now
